### PR TITLE
Remove index sections on apps and Metal.

### PR DESCRIPTION
**Pull request by blakejohnson**
Removes sections on Apps modules (Nature, Finance, Optimization, and Machine Learning) and Qiskit Metal from the documentation index page.

### CODE REVIEW

**Review by Eric-Arellano — COMMENTED** [review]
I'm personally a +1 for this. We need to redo the home page anyways to better clarify what is Qiskit: e.g. https://github.com/Qiskit/qiskit-metapackage/issues/1646.

File: docs/index.rst [hunk 1]
```diff
@@ -90,26 +54,6 @@ Interested in running experiments on real qubits?
...
-        :columns: auto
-
-        Qiskit Metal
-        ^^^^^^^^^^^^^^
-
-        .. image:: images/metal.png
-          :scale: 35 %
-          :align: center
-          :target: https://qiskit.org/ecosystem/metal/
-
```
Comment (jakelishman) [inline docs/index.rst]: I think with this removal, there shouldn't be any more references to `images/metal.png`, so we can probably remove that too.

Comment (blakejohnson) [reply docs/index.rst]: Removed in [9340851](https://github.com/Qiskit/qiskit-metapackage/pull/1759/commits/9340851fd1e76c44f3484da4e806ff3bfb220e1b)

File: docs/index.rst [hunk 2]
```diff
@@ -45,42 +45,6 @@ around the world by beginners, hobbyists, educators, researchers, and commercial
...
-  :header: Qiskit Optimization
-  :button_link: https://qiskit.org/ecosystem/optimization/
-  :button_text: Qiskit Optimization
-
-.. raw:: html
-
-  </div>
-
 Interested in running experiments on real qubits?
 #################################################
```
Comment (Eric-Arellano) [inline docs/index.rst]: Why keep this one? That's not a rhetorical question - I'm not sure what makes sense to keep vs remove.

Comment (blakejohnson) [reply docs/index.rst]: I think we should have sections for the primary repos in qiskit-extensions. So, I would propose actually adding tiles for Quantum Serverless and Circuit Knitting Toolbox.

Comment (1ucian0) [reply docs/index.rst]: in https://github.com/Qiskit/qiskit-metapackage/pull/1759/commits/5ee6acb4811ed091cfd312edded54eef9183c8bc, I added those in the main tier. dynamics, experiments, provider, and runtime.  I also removed the image.

File: docs/index.rst [hunk 3]
```diff
@@ -1,114 +1,75 @@
-.. raw:: html
-
-  <div style="width: 100%; align: center">
-
-.. raw:: html
-  :file: images/qiskit_main.svg
```
Comment (Eric-Arellano) [inline docs/index.rst:6]: This SVG should be deleted, right?

File: docs/index.rst [hunk 4]
```diff
@@ -1,114 +1,75 @@
...
 .. raw:: html
 
-  <div class="tutorials-callout-container">
-     <div class="row">
+  <div class="tutorials-callout-container"><div class="row">
+
 
 .. qiskit-call-to-action-item::
   :description: Find out which Qiskit providers support execution on real quantum services.
   :header: Access to quantum systems
```
Comment (Eric-Arellano) [inline docs/index.rst:21]: Nit, maybe move the `:header:` option to the top of each `qiskit-call-to-action-item`. It makes no difference for the final rendered docs. But it makes it easier for humans to work with this file to see what each Call To Action is for.

File: docs/index.rst [hunk 5]
```diff
@@ -1,114 +1,75 @@
...
+
 .. raw:: html
 
-  <div class="tutorials-callout-container">
-     <div class="row">
+  <div class="tutorials-callout-container"><div class="row">
+
 
 .. qiskit-call-to-action-item::
   :description: Find out which Qiskit providers support execution on real quantum services.
```
Comment (Eric-Arellano) [inline docs/index.rst]:
```suggestion
  :description: Qiskit providers offer execution on real quantum services.
```

File: docs/index.rst [hunk 6]
```diff
@@ -1,114 +1,75 @@
...
-######################################
+.. qiskit-call-to-action-item::
+  :description: Tools for building and solving models of quantum systems in Qiskit
+  :header: Qiskit Dynamics
+  :button_link: https://qiskit.org/ecosystem/dynamics/
+  :button_text: Qiskit Dynamics documentation
 
-   .. grid:: 2
+.. qiskit-call-to-action-item::
+  :description: Qiskit Runtime are a cloud base implementation of the Qiskit primitives to effectively execute workloads on IBM Quantum systems.
```
Comment (Eric-Arellano) [inline docs/index.rst]:
```suggestion
  :description: Qiskit Runtime is a cloud base implementation of the Qiskit primitives to effectively execute workloads on IBM Quantum systems.
```

File: docs/index.rst [hunk 7]
```diff
@@ -1,114 +1,75 @@
...
+  :header: Qiskit IBM Runtime
+  :button_link:  https://qiskit.org/ecosystem/ibm-runtime/
+  :button_text: Qiskit Runtime documentation
 
-    .. grid-item-card::
-        :columns: auto
+.. qiskit-call-to-action-item::
+  :description: Qiskit provider that allows accessing the IBM Quantum systems and cloud simulators.
```
Comment (Eric-Arellano) [inline docs/index.rst]:
```suggestion
  :description: A Qiskit provider that allows accessing the IBM Quantum systems and cloud simulators.
```

**Review by Eric-Arellano — COMMENTED** [review]
Nice!

File: docs/index.rst (hunk 3 above)
Comment (1ucian0) [reply docs/index.rst:6]: removing it in 25058f7ab95a5723a85f065b5f29d80c4e9a6263

File: docs/index.rst (hunk 4 above)
Comment (1ucian0) [reply docs/index.rst:21]: good point. Done in b06179ccb94ec78d7adfaa245b70b95d5eb2ca4f

**Review by Eric-Arellano — APPROVED** [review]
Thanks!

**Pull request merged**